src/components/TableHeader/TableHeader.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import './TableHeader.scss';
+
+import {
+  THEAD_TITLE, sortingList,
+} from '../../api/helper';
+import { Person } from '../../api/interface';
+
+type TableHeaderT = {
+  people: Person[];
+  setPeople: Function;
+  isAddingForm: boolean;
+};
+
+export const TableHeader: React.FC<TableHeaderT> = ({ people, setPeople, isAddingForm }) => {
+  const [sortOrder, setSortOrder] = useState({
+    name: 'both',
+    sex: 'both',
+    born: 'both',
+    died: 'both',
+  });
+  const history = useHistory();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const sortPeopleList = useCallback(() => {
+    setPeople(
+      sortingList(
+        people,
+        searchParams.get('sortBy') || '',
+        searchParams.get('sortOrder') || '',
+      ),
+    );
+  }, [searchParams, history]);
+
+  useEffect(() => {
+    if (searchParams.has('sortBy')) {
+      searchParams.set(
+        'sortOrder',
+        (sortOrder as never)[searchParams.get('sortBy')?.toLowerCase() || ''],
+      );
+      history.push(`?${searchParams.toString()}`);
+      sortPeopleList();
+    }
+  }, [sortOrder]);
+
+  const handleSortList = (event: React.MouseEvent<HTMLTableRowElement>) => {
+    const { textContent } = (event.target as HTMLElement);
+
+    if (textContent !== THEAD_TITLE.father && textContent !== THEAD_TITLE.mother) {
```

Comment:
THEAD_TITLE может содержать в себе объекты, у которых есть поле isSortable. У mother и father это поле будет false, и тут можно будет написать просто
```suggestion
    if (THEAD_TITLE[textContent].sortable) {
```